### PR TITLE
General | Basic for save system

### PR DESCRIPTION
**Pull request by L5478**
- Save system
- Save classes for dialogues and inventory

### CODE REVIEW

File: Assets/_Scenes/General/Scripts/Save/SaveSystem.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+    private void Save()
+    {
+        // Save all states
+        DialogueSave.instance.SaveDialogueStates();
+        InventorySave.instance.SaveInventoryState();
+
+
+        // Save everything to the binary file
+        string destination = Application.persistentDataPath + "/dialoguestates.fuck";
```
Comment (RyyzQ) [inline Assets/_Scenes/General/Scripts/Save/SaveSystem.cs]: 🤔

File: Assets/_Scenes/General/Scripts/Save/DialogueSave.cs [hunk 2]
```diff
@@ -16,6 +16,7 @@ private void Awake()
         instance = this;
     }
 
+    [HideInInspector]
     public Dictionary<string, int> DialogueStates = new Dictionary<string, int>();
```
Comment (RyyzQ) [inline Assets/_Scenes/General/Scripts/Save/DialogueSave.cs]: Does this need to be public?

File: Assets/_Scenes/General/Scripts/Save/SaveSystem.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            Debug.LogWarning("More than one " + instance + " found");
+            Destroy(instance);
+        }
+
+        instance = this;
+    }
+
+    void Update()
+    {
+        if (Input.GetKeyDown(KeyCode.K))
```
Comment (RyyzQ) [inline Assets/_Scenes/General/Scripts/Save/SaveSystem.cs]: If this is debugging code, should probably be removed. If not, should make proper inputs for this in the input settings, so a player on a controller can use this as well.

(If you're already cleaning this, feel free to ignore)

File: Assets/_Scenes/General/Scripts/Save/DialogueSave.cs (hunk 2 above)
Comment (L5478) [reply Assets/_Scenes/General/Scripts/Save/DialogueSave.cs]: Yes

File: Assets/_Scenes/General/Scripts/Save/SaveSystem.cs (hunk 3 above)
Comment (L5478) [reply Assets/_Scenes/General/Scripts/Save/SaveSystem.cs]: Those are just for testing purposes and I added comment there as well. We will keep those for now, but of course those will be changed afterwards.

File: Assets/_Scenes/General/Scripts/Save/DialogueSave.cs (hunk 2 above)
Comment (L5478) [reply Assets/_Scenes/General/Scripts/Save/DialogueSave.cs]: SaveSystem is calling those

**Pull request merged**